physics-informed-neural-networks-for-steady-cavity-flow/README.md

```diff
@@ -207,25 +199,21 @@ yeq = xyEquation(:, 2);
 [psi, p] = forward(net, xeq, yeq);
 
 % Compute gradients.
-psisum = sum(psi,1);
-u = dlgradient(psisum, yeq, EnableHigherDerivatives=true);
-v = -1.*dlgradient(psisum, xeq, EnableHigherDerivatives=true);
-
-usum = sum(u,1);
-ux = dlgradient(usum, xeq, EnableHigherDerivatives=true);
-uy = dlgradient(usum, yeq, EnableHigherDerivatives=true);
-uxx = dlgradient(sum(ux,1), xeq, EnableHigherDerivatives=true);
-uyy = dlgradient(sum(uy,1), yeq, EnableHigherDerivatives=true);
-
-vsum = sum(v,1);
-vx = dlgradient(vsum, xeq, EnableHigherDerivatives=true);
-vy = dlgradient(vsum, yeq, EnableHigherDerivatives=true);
-vxx = dlgradient(sum(vx,1), xeq, EnableHigherDerivatives=true);
-vyy = dlgradient(sum(vy,1), yeq, EnableHigherDerivatives=true);
-
-psum = sum(p,1);
-px = dlgradient(psum, xeq, EnableHigherDerivatives=true);
-py = dlgradient(psum, yeq, EnableHigherDerivatives=true);
+u = dljacobian(psi', yeq, 1);
```

Comment:
If `psi` and `yeq` are both "batch x features" I think `u = dljacobian(psi, yeq, 2)` would work here. But then `u` is "hidden x batch x features" where `hidden==size(psi,2)`, and since you end up wanting more derivatives along that "hidden" dimension, I guess you have to permute or swap to `dljacobian(u,xeq,1)` anyway. I think having stuff as "features x batch" works out more nicely for me typically, which means replacing `featureInputLayer` with `inputLayer` if you want to use unlabelled `dlarray`.

Reply:
I can see how working in "features x batch" is most convenient for the network. On the other hand, the "batch x features" layout is closer to the original data layout. 

I'll keep things as they are for now and can switch to "features x batch" later if we think it is neater.